Fix for https://relationalai.atlassian.net/browse/RAI-9265 - Failed to create user as it already exists

miazamrai opened the pull request:
Fix for https://relationalai.atlassian.net/browse/RAI-9265
Since we were using a common/fixed email address and all the CI/CD workflows use the same account, user-service failed to create the user with fixed email address again because another workflow had already created it in parallel.

larf311 commented:
The change looks good but now I'm worried we'll somehow end up with a bunch of users if the tests fail to delete them for some reason. Any thoughts on how we can prevent that? Or at least be properly alerted?

miazamrai commented:
> The change looks good but now I'm worried we'll somehow end up with a bunch of users if the tests fail to delete them for some reason. Any thoughts on how we can prevent that? Or at least be properly alerted?

User deletion is happening in the tear down. If it fails to delete, then we can return an error exit code from the tear down function so that the test marked as failed. This is one way of tracking it.

miazamrai commented:
> > The change looks good but now I'm worried we'll somehow end up with a bunch of users if the tests fail to delete them for some reason. Any thoughts on how we can prevent that? Or at least be properly alerted?
> 
> User deletion is happening in the tear down. If it fails to delete, then we can return an error exit code from the tear down function so that the test marked as failed. This is one way of tracking it.

Also DeleteUser is happening in the TestUser itself. If that fails the test will fail and so do the workflow.

miazamrai commented:
@larf311 I have added the functionality in the teardown function to return an exit code but then I had to change the way delete database and delete engine was happening. If the tests had already deleted the database or engine the existing teardown function would return a wrong exit code. So, I have added to code to find and delete as it is doing for user and oauth client.
